internal/bminventory/inventory_v2_handlers.go

```diff
@@ -605,13 +607,66 @@ func (b *bareMetalInventory) signURL(ctx context.Context, infraEnvID, urlString,
 	return urlString, &expiresAt, nil
 }
 
-const ipxeScriptFormat = `#!ipxe
+const ipxeFirstScriptFormat = `#!ipxe
+chain %s/%s?file_name=ipxe-script&mac=${net0/mac}
+`
+
+const ipxeSecondScriptFormat = `#!ipxe
 initrd --name initrd %s
 kernel %s initrd=initrd coreos.live.rootfs_url=%s random.trust_cpu=on rd.luks.options=discard ignition.firstboot ignition.platform.id=metal console=tty1 console=ttyS1,115200n8 coreos.inst.persistent-kargs="console=tty1 console=ttyS1,115200n8"
 boot
 `
 
-func (b *bareMetalInventory) infraEnvIPXEScript(ctx context.Context, infraEnv *common.InfraEnv) (string, error) {
+func (b *bareMetalInventory) firstIPXEScript(urlPath string) (string, error) {
+	serviceURL, err := url.Parse(b.ServiceBaseURL)
+	if err != nil {
+		return "", err
+	}
+	if b.httpListenPortStr != "" {
+		httpListenPort, err := strconv.ParseInt(b.httpListenPortStr, 10, 64)
+		if err != nil {
+			return "", err
+		}
+		serviceURL.Host = fmt.Sprintf("%s:%d", serviceURL.Hostname(), httpListenPort)
+	}
+	serviceURL.Scheme = "http"
+
+	return fmt.Sprintf(ipxeFirstScriptFormat, serviceURL.String(), urlPath), nil
+}
+
+func (b *bareMetalInventory) shouldServScript(infraEnv *common.InfraEnv, mac *strfmt.MAC) (bool, error) {
+	var hosts []*models.Host
+	macStr := mac.String()
+	if err := b.db.Where("infra_env_id = ? and (inventory like ? or inventory like ?)", infraEnv.ID.String(), fmt.Sprintf("%%%s%%", strings.ToUpper(macStr)),
+		fmt.Sprintf("%%%s%%", strings.ToLower(macStr))).Find(&hosts).Error; err != nil {
```

Comment:
This is quite the query.

I wonder if this is worth its own function.
Also is this specifically better than doing `LOWER(inventory) like ...`?
Then you wouldn't need the second case.

Reply:
The inventory is the field we query. mac is the variable we pass to the query.  
I don't think to perform LOWER on such a big field is a good idea.